src/librustc_mir/transform/check_unsafety.rs

```diff
@@ -624,16 +651,20 @@ pub fn check_unsafety(tcx: TyCtxt<'_>, def_id: DefId) {
     let UnsafetyCheckResult { violations, unsafe_blocks } =
         tcx.unsafety_check_result(def_id.expect_local());
 
+    let or_block_msg = if tcx.features().unsafe_block_in_unsafe_fn { "" } else { " or block" };
```

Comment:
I don't think checking the feature here is the right trigger. You need to check if the `unsafe_op_in_unsafe_fn` lint is enabled here.